test/generic.spec.js

```diff
@@ -0,0 +1,174 @@
+var fs = require('fs');
+var Promise = require('bluebird');
+var readFile = Promise.promisify(require('fs').readFile);
+var expect = require('chai').expect;
+var async = require('async');
+var Converter = require('csvtojson').Converter;
+var pg = require('pg');
+
+/*
+ This enforce driver to cast bigint (64-bits length) values into javascript (32-bit) integer.
+ We advice to do not use bigint data type. However this setting is useful for COUNT aggregations that
+ returns bigint.
+ Plese refer https://github.com/brianc/node-postgres/wiki/pg#pgdefaultsparseint8
+ */
```

Comment:
http://ecma262-5.com/ELS5_HTML.htm#Section_8.5

To be honest, I don't know current specs for ECMA6 but in 5 integer was definitely defined as 64 bit and that would fix your problems, are you absolutely sure that it is 32 bit?